pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -214,24 +219,38 @@ func getNodeUsage(
 		if len(node.Status.Allocatable) > 0 {
 			nodeCapacity = node.Status.Allocatable
 		}
+		lowResourceThreshold := map[v1.ResourceName]*resource.Quantity{
+			v1.ResourceCPU:    resource.NewMilliQuantity(int64(float64(lowThreshold[v1.ResourceCPU])*float64(nodeCapacity.Cpu().MilliValue())*0.01), resource.DecimalSI),
+			v1.ResourceMemory: resource.NewQuantity(int64(float64(lowThreshold[v1.ResourceMemory])*float64(nodeCapacity.Memory().Value())*0.01), resource.BinarySI),
+			v1.ResourcePods:   resource.NewQuantity(int64(float64(lowThreshold[v1.ResourcePods])*float64(nodeCapacity.Pods().Value())*0.01), resource.DecimalSI),
+		}
+		for name, percent := range lowThreshold {
+			if !isBasicResource(name) {
+				cap := nodeCapacity[name]
+				lowResourceThreshold[name] = resource.NewQuantity(int64(float64(percent)*float64(cap.Value())*0.01), resource.DecimalSI)
+			}
+		}
+		highResourceThreshold := map[v1.ResourceName]*resource.Quantity{
+			v1.ResourceCPU:    resource.NewMilliQuantity(int64(float64(highThreshold[v1.ResourceCPU])*float64(nodeCapacity.Cpu().MilliValue())*0.01), resource.DecimalSI),
+			v1.ResourceMemory: resource.NewQuantity(int64(float64(highThreshold[v1.ResourceMemory])*float64(nodeCapacity.Memory().Value())*0.01), resource.BinarySI),
+			v1.ResourcePods:   resource.NewQuantity(int64(float64(highThreshold[v1.ResourcePods])*float64(nodeCapacity.Pods().Value())*0.01), resource.DecimalSI),
+		}
+		for name, percent := range highThreshold {
+			if !isBasicResource(name) {
+				cap := nodeCapacity[name]
+				highResourceThreshold[name] = resource.NewQuantity(int64(float64(percent)*float64(cap.Value())*0.01), resource.DecimalSI)
+			}
+		}
 
 		nodeUsageList = append(nodeUsageList, NodeUsage{
 			node:    node,
-			usage:   nodeUtilization(node, pods),
+			usage:   nodeUtilization(node, pods, lowResourceThreshold),
```

Comment:
Instead of passing `lowResourceThreshold` I'd rather pass a list of all resources which are taken into account (including extended resources). Can you create a function that will take `thresholds` (or `targetThresholds` since both variables have to have the same list of resources after validation) and produces the list? E.g. `func getResourceNames(api.ResourceThresholds) []v1.ResourceName`.

Reply:
Sure, I have just added func `getResourceNames` in latest squashed commits, and help review again please.